sdk/aqueduct/backend/api_client.py

```diff
@@ -272,6 +275,86 @@ def connect_integration(
+        self,
+        dag: DAG,
+    ) -> Dict[str, EngineStatusResponse]:
+        """Makes a request against the /api/integration/engine/status endpoint.
+
+        Args:
+            dag:
+                The DAG object. We will extract the engine integration IDs and send them
+                to the backend to retrieve their status. Currently, we are only interested in
+                the status of dynamic engines.
+
+        Returns:
+            A EngineStatusResponse object, parsed from the backend endpoint's response.
+        """
+        engine_integration_ids = set()
+
+        dag_engine_config = dag.engine_config
+        if dag_engine_config.type == RuntimeType.K8S:
+            assert dag_engine_config.k8s_config is not None
+            engine_integration_ids.add(str(dag_engine_config.k8s_config.integration_id))
+        for op in dag.operators.values():
+            if op.spec.engine_config and op.spec.engine_config.type == RuntimeType.K8S:
+                assert op.spec.engine_config.k8s_config is not None
+                engine_integration_ids.add(str(op.spec.engine_config.k8s_config.integration_id))
+
+        return self.get_engine_status(list(engine_integration_ids))
+
+    def get_engine_status(
+        self,
+        engine_integration_ids: List[str],
+    ) -> Dict[str, EngineStatusResponse]:
+        """Makes a request against the /api/integration/engine/status endpoint.
+
+        Args:
+            engine_integration_ids:
+                A list of engine integration IDs. Currently, we are only interested in
+                the status of dynamic engines.
+
+        Returns:
+            A EngineStatusResponse object, parsed from the backend endpoint's response.
+        """
+        headers = self._generate_auth_headers()
+        headers["integration-ids"] = json.dumps(engine_integration_ids)
+
+        url = self.construct_full_url(self.GET_DYNAMIC_ENGINE_STATUS_ROUTE)
+        resp = requests.get(url, headers=headers)
+        self.raise_errors(resp)
+
+        return {
+            engine_status["name"]: EngineStatusResponse(**engine_status)
+            for engine_status in resp.json()
+        }
+
+    def modify_engine(
+        self,
+        action: str,
+        integration_id: str,
+    ) -> None:
+        """Makes a request against the /api/integration/engine/{integrationId} endpoint.
+
+        Args:
+            integration_id:
+                The engine integration ID.
+        """
+        headers = self._generate_auth_headers()
+        headers["action"] = action
+
+        url = self.construct_full_url(self.MODIFY_DYNAMIC_ENGINE_ROUTE_TEMPLATE % integration_id)
+
+        if action == "create" or action == "delete":
+            resp = requests.post(url, headers=headers)
+        else:
+            raise InvalidRequestError(
+                "Invalid action %s for interacting with dynamic engine." % action
+            )
+
+        self.raise_errors(resp)
+        return
```

Comment:
nit: I think it's fine to leave out the return line which is what we usually do.